src/ray/core_worker/task_manager.cc

```diff
@@ -304,12 +309,24 @@ void TaskManager::CompletePendingTask(const TaskID &task_id,
     if (task_retryable) {
       // Pin the task spec if it may be retried again.
       release_lineage = false;
+      it->second.lineage_footprint_bytes = it->second.spec.GetMessage().ByteSizeLong();
+      total_lineage_footprint_bytes_ += it->second.lineage_footprint_bytes;
+      if (total_lineage_footprint_bytes_ > max_lineage_bytes_) {
+        RAY_LOG(INFO) << "Evicting task lineage. Total lineage size is "
+                      << total_lineage_footprint_bytes_ / 1e6
+                      << "MB, which exceeds the limit of " << max_lineage_bytes_ / 1e6
+                      << "MB";
+        evict_all_lineage = true;
+      }
     } else {
       submissible_tasks_.erase(it);
     }
   }
 
   RemoveFinishedTaskReferences(spec, release_lineage, worker_addr, reply.borrowed_refs());
+  if (evict_all_lineage) {
```

Comment:
Is it intended we evict all lineage when we reach to the limit? Why not evicting it in FIFO or LRU order?

Reply:
Mostly just to keep it simple for now, but let me see how easy it would be to do this.